test/show.jl

```diff
@@ -818,3 +811,20 @@ end
     @test sprint(show, Val(:Float64)) == "Val{:Float64}()" # Val of a symbol
     @test sprint(show, Val(true))     == "Val{true}()"     # Val of a value
 end
+
+@testset "printing of Pair's" begin
+    for (p, s) in (Pair(1.0,2.0)                          => "1.0 => 2.0",
+                   Pair(Pair(1,2), Pair(3,4))             => "(1=>2) => (3=>4)",
+                   Pair{Integer,Int64}(1, 2)              => "Pair{Integer,Int64}(1, 2)",
+                   (Pair{Integer,Int64}(1, 2) => 3)       => "(1=>2) => 3",
```

Comment:
I mean that the summary of the array already indicates an `eltype` of `Pair{Int64,Any}`, so each element woudn't really need to repeat this information. But I realize it's no different from e.g. `[Integer[1, 2]]`, so not specific to pairs.